sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -0,0 +1,288 @@
+        defaultResponse
+      case proto.PipelineCommand.CommandTypeCase.DEFINE_SQL_GRAPH_ELEMENTS =>
+        logInfo(s"Register sql datasets cmd received: $cmd")
+        defineSqlGraphElements(cmd.getDefineSqlGraphElements, sparkSession)
+        defaultResponse
+      case other => throw new UnsupportedOperationException(s"$other not supported")
+    }
+  }
+
+  private def createDataflowGraph(cmd: proto.PipelineCommand.CreateDataflowGraph): String = {
+    val defaultCatalog = Option
+      .when(cmd.hasDefaultCatalog)(cmd.getDefaultCatalog)
+      .getOrElse {
+        logInfo(
+          s"No default catalog was supplied. Falling back to the session catalog `spark_catalog`).")
+        "spark_catalog"
+      }
+
+    val defaultDatabase = Option
+      .when(cmd.hasDefaultDatabase)(cmd.getDefaultDatabase)
+      .getOrElse {
+        logInfo(s"No default catalog was supplied. Falling back to `default`).")
+        "default"
+      }
+
+    val defaultSqlConf = cmd.getSqlConfMap.asScala.toMap
+
+    DataflowGraphRegistry.createDataflowGraph(
+      defaultCatalog = defaultCatalog,
+      defaultDatabase = defaultDatabase,
+      defaultSqlConf = defaultSqlConf)
+  }
+
+  private def defineSqlGraphElements(
+      cmd: proto.PipelineCommand.DefineSqlGraphElements,
+      session: SparkSession): Unit = {
+    val dataflowGraphId = cmd.getDataflowGraphId
+
+    val graphElementRegistry = DataflowGraphRegistry.getDataflowGraphOrThrow(dataflowGraphId)
+    val sqlGraphElementRegistrationContext = new SqlGraphRegistrationContext(graphElementRegistry)
+    sqlGraphElementRegistrationContext.processSqlFile(cmd.getSqlText, cmd.getSqlFilePath, session)
+  }
+
+  private def defineDataset(
+      dataset: proto.PipelineCommand.DefineDataset,
+      sparkSession: SparkSession): Unit = {
+    val dataflowGraphId = dataset.getDataflowGraphId
+    val graphElementRegistry = DataflowGraphRegistry.getDataflowGraphOrThrow(dataflowGraphId)
+
+    dataset.getDatasetType match {
+      case proto.DatasetType.MATERIALIZED_VIEW | proto.DatasetType.TABLE =>
+        val tableIdentifier =
+          GraphIdentifierManager.parseTableIdentifier(dataset.getDatasetName, sparkSession)
+        graphElementRegistry.registerTable(
+          Table(
+            identifier = tableIdentifier,
+            comment = Option(dataset.getComment),
+            specifiedSchema = Option.when(dataset.hasSchema)(
+              DataTypeProtoConverter
+                .toCatalystType(dataset.getSchema)
+                .asInstanceOf[StructType]),
+            partitionCols = Option(dataset.getPartitionColsList.asScala.toSeq)
+              .filter(_.nonEmpty),
+            properties = dataset.getTablePropertiesMap.asScala.toMap,
+            baseOrigin = QueryOrigin(
+              objectType = Option(QueryOriginType.Table.toString),
+              objectName = Option(tableIdentifier.unquotedString),
+              language = Option(Python())),
```

Comment:
Python? You technically don't know that. To what end do we need to record this information.

Reply:
Currently all the SQL code goes through `defineSqlGraphElement` so anything going through this path is Python. However, it's not being used right now so I'm happy to remove it